src/peer_bootstrap_node.cpp

```diff
@@ -25,40 +50,75 @@ void PeerBootstrapNode::sendData(const std::vector<uint8_t> &data)
     // Data sending code here
 }
 
-void PeerBootstrapNode::receiveData(std::vector<uint8_t> &data)
+void PeerBootstrapNode::receiveData(const asio::error_code &error, std::size_t bytes_transferred)
 {
-    Logger::log(LogLevel::DEBUG, "Receiving data...");
-    // Data receiving code here
-}
+    if (!error && bytes_transferred > 0)
+    {
+        std::vector<uint8_t> data(receive_buffer_.begin(), receive_buffer_.begin() + bytes_transferred);
+        Logger::log(LogLevel::INFO, "Received data from " + remote_endpoint_.address().to_string() + ":" +
+                                        std::to_string(remote_endpoint_.port()) +
+                                        " Size: " + std::to_string(bytes_transferred));
+
+        // Process the data (placeholder)
+        Logger::log(LogLevel::DEBUG, "Data: " + std::string(data.begin(), data.end()));
 
+        // Continue listening
+        socket_.async_receive_from(
+            asio::buffer(receive_buffer_), remote_endpoint_,
+            [this](const asio::error_code &error, std::size_t bytes_transferred)
+            {
+                receiveData(error, bytes_transferred);
+            });
+    }
+    else
+    {
+        Logger::log(LogLevel::ERROR, "Receive error: " + error.message());
+    }
+}
+void PeerBootstrapNode::startListening()
+{
+    if (listening_)
+    {
+        Logger::log(LogLevel::ERROR, "Already listening for incoming data.");
+        return;
+    }
+    listening_ = true;
+    Logger::log(LogLevel::INFO, "Listening for incoming data on " + host_ip_ + ":" + std::to_string(host_port_));
+    socket_.async_receive_from(asio::buffer(receive_buffer_), remote_endpoint_,
+                               [this](const asio::error_code &error, std::size_t bytes_transferred)
+                               {
+                                   receiveData(error, bytes_transferred);
+                               });
+    io_thread_ = std::thread([this]
+                             { io_context_.run(); });
+}
 int main(int argc, char *argv[])
 {
     if (argc != 3)
     {
+        Logger::setLogFile("peer_bootstrap_node.log");
         Logger::log(LogLevel::ERROR, "Usage: ./peer_bootstrap <host_ip> <host_port>");
         return 1;
     }
-
     std::string host_ip = argv[1];
     uint16_t host_port = std::stoi(argv[2]);
     uint32_t id = 12345;
+    Logger::setLogFile("peer_bootstrap_node.log");
     PeerBootstrapNode bootstrapNode(id);
-    bootstrapNode.setHostAddress(host_ip, host_port);
-    bootstrapNode.connect();
-
-    bool running = true;
-    while (running)
+    try
     {
-        std::vector<uint8_t> data;
-        bootstrapNode.receiveData(data);
+        bootstrapNode.setHostAddress(host_ip, host_port);
+        bootstrapNode.connect();
```

Comment:
Still don't like this name

Reply:
It's how it is called theoretically, I also would probably prefer discovery node or something, but it is what it is.